ports/esp32/esp32_rmt.c

```diff
@@ -61,13 +64,25 @@ STATIC mp_obj_t esp32_rmt_make_new(const mp_obj_type_t *type, size_t n_args, siz
         { MP_QSTR_id,        MP_ARG_REQUIRED | MP_ARG_INT, {.u_int = -1} },
         { MP_QSTR_pin,       MP_ARG_REQUIRED | MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = mp_const_none} },
         { MP_QSTR_clock_div,                   MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 8} }, // 100ns resolution
+        { MP_QSTR_carrier_duty_percent,        MP_ARG_KW_ONLY | MP_ARG_INT, {.u_int = 50} },
+        { MP_QSTR_carrier_freq_hz,             MP_ARG_KW_ONLY | MP_ARG_OBJ, {.u_obj = MP_OBJ_NULL} },
```

Comment:
following uses of frequency elsewhere, this doesn't need a `_hz` suffix (frequency is always in Hz in MicroPython); so it can just be `carrier_freq`

Reply:
Why is this an obj, not an int?

Reply:
It's so the code can detect between "not supplied" and "supplied".  If it's not supplied then the carrier freq is disabled.  Alternative would be to make -1 indicate disabled (and be the default value), and actually I think that would be a better way to do it so the user can explicitly select disabled by passing -1.

Reply:
Or zero. A freq of zero is DC...

Reply:
> Or zero. A freq of zero is DC...

Yes!

Reply:
OK I changed so its enabled with an int greater than zero, default is zero